### PR TITLE
provide an API to scan all images, and some refactory

### PR DESCRIPTION
**Pull request by reasonerjt**

### CODE REVIEW

File: src/common/utils/registry/repository.go [hunk 1]
```diff
@@ -104,13 +105,22 @@ func (r *Repository) ListTag() ([]string, error) {
 		}{}
 
 		if err := json.Unmarshal(b, &tagsResp); err != nil {
+			sort.Strings(tags)
```
Comment (ywk253100) [inline src/common/utils/registry/repository.go]: Move this after the if statement.

File: src/ui/api/repository.go [hunk 2]
```diff
@@ -782,6 +757,25 @@ func (ra *RepositoryAPI) VulnerabilityDetails() {
...
+	}
+	if !ra.SecurityCtx.IsAuthenticated() {
+		ra.HandleUnauthorized()
+		return
+	}
+	if !ra.SecurityCtx.IsSysAdmin() {
+		ra.HandleForbidden(ra.SecurityCtx.GetUsername())
+		return
+	}
+	uiutils.ScanAllImages()
```
Comment (ywk253100) [inline src/ui/api/repository.go]: Check the error

File: src/ui/api/utils.go [hunk 3]
```diff
@@ -316,7 +307,7 @@ func diffRepos(reposInRegistry []string, reposInDB []string,
 			if err != nil {
 				return needsAdd, needsDel, err
 			}
-			client, err := NewRepositoryClient(endpoint, true,
+			client, err := uiutils.NewRepositoryClientForUI(endpoint, true,
 				"admin", repoInR, "repository", repoInR, "pull")
```
Comment (ywk253100) [inline src/ui/api/utils.go]: Should be uiutils.NewRepositoryClientForUI(endpoint, true,"admin", repoInR, "pull")

File: src/ui/api/repository.go (hunk 2 above)
Comment (reasonerjt) [inline src/ui/api/repository.go]: thanks

File: src/common/utils/registry/repository.go (hunk 1 above)
Comment (reasonerjt) [inline src/common/utils/registry/repository.go]: thanks

**Pull request merged**